### PR TITLE
Tasks can set allowunrestrictednetworkaccess if they like

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/services/pipeline/task.http.go [hunk 1]
```diff
@@ -14,11 +14,46 @@ import (
 	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
+type (
```
Comment (connorwstein) [inline core/services/pipeline/task.http.go:17]: should we put this in the null package?

File: core/services/pipeline/runner_test.go [hunk 2]
```diff
@@ -164,6 +166,7 @@ func TestRunner(t *testing.T) {
 		for _, run := range runs {
 			if run.DotID() == "ds1" {
 				assert.True(t, run.Error.IsZero())
+				require.NotNil(t, resp, run.Output)
```
Comment (connorwstein) [inline core/services/pipeline/runner_test.go:169]: Could we also assert somewhere that the AllowUnrestrictedNetworkAccess on the HTTPTask is actually set to true from the DOT source?

File: core/services/pipeline/task.http.go (hunk 1 above)
Comment (samsondav) [reply core/services/pipeline/task.http.go:17]: Well, there is already a `null.Bool` but sadly it didn't work due to incompatibilities with mapstructure, that's why I opted to keep it scoped to this package.

File: core/services/pipeline/runner_test.go (hunk 2 above)
Comment (samsondav) [reply core/services/pipeline/runner_test.go:169]: I could not figure out an easy or obvious way to test this.

Perhaps @spooktheducks has some ideas

Comment (connorwstein) [reply core/services/pipeline/runner_test.go:169]: fwiw I ended up doing something similar here https://github.com/smartcontractkit/chainlink/pull/3698/files#diff-090f788b3a083e95e45cee241250eeb3cb0215d4b553c2737753634ab9ef4b79R12

**Review by connorwstein — APPROVED** [review]

**Pull request merged**